pkg/controller/logstorage/linseed.go

```diff
@@ -44,6 +44,7 @@ func (r *ReconcileLogStorage) createLinseed(
 	managementCluster bool,
 	usePSP bool,
 	esClusterConfig *relasticsearch.ClusterConfig,
+	hasDPIResource bool,
```

Comment:
Rather than pass this in, you should just have the client query this from within this function.

We have way too many arguments here, and really the only things that should be arguments are variables that are shared between multiple components.